pinot-spi/src/main/java/org/apache/pinot/spi/config/table/TableConfig.java

```diff
@@ -53,6 +53,7 @@ public class TableConfig extends BaseJsonConfig {
   public static final String INGESTION_CONFIG_KEY = "ingestionConfig";
   public static final String TIER_CONFIGS_LIST_KEY = "tierConfigs";
   public static final String TUNER_CONFIG_LIST_KEY = "tunerConfigs";
+  public static final String INSTANCE_PARTITIONS_MAP_CONFIG_KEY = "instancePartitionsMap";
```

Comment:
(minor) Put it next to `INSTANCE_ASSIGNMENT_CONFIG_MAP_KEY` since they are both for the instance assignment. Same for the order of variable declaration, argument, getter, setter